services/src/main/java/org/fao/geonet/services/metadata/Update.java

```diff
@@ -132,6 +146,47 @@ public Element serviceSpecificExec(Element params, ServiceContext context) throw
 			dataMan.endEditingSession(id, session);
 		}
 
+		if (finished && !forget && !commit) {
+			SettingManager sm = gc.getBean(SettingManager.class);
+
+			boolean forceValidationOnMdSave = sm.getValueAsBool("metadata/workflow/forceValidationOnMdSave");
+
+			// Save validation if the forceValidationOnMdSave is enabled
+			if (forceValidationOnMdSave && !showValidationErrors.equals("true")) {
+				final MetadataRepository metadataRepository = gc.getBean(MetadataRepository.class);
+				Metadata metadata = metadataRepository.findOne(id);
+
+				dataMan.doValidate(metadata.getDataInfo().getSchemaId(), metadata.getId() + "",
+						new Document(metadata.getXmlData(false)), context.getLanguage());
+				dataMan.indexMetadata(id, true);
+			}
+
+			boolean automaticUnpublishInvalidMd = sm.getValueAsBool("metadata/workflow/automaticUnpublishInvalidMd");
+
+			// Unpublish the metadata automatically if the setting automaticUnpublishInvalidMd is enabled and
+			// the metadata becomes invalid
+			if (automaticUnpublishInvalidMd) {
+				final OperationAllowedRepository operationAllowedRepo = context.getBean(OperationAllowedRepository.class);
+
+				boolean isPublic = (operationAllowedRepo.count(where(hasMetadataId(id)).and(hasOperation(ReservedOperation
+						.view)).and(hasGroupId(ReservedGroup.all.getId()))) > 0);
+
+				if (isPublic) {
+					final MetadataValidationRepository metadataValidationRepository = gc.getBean(MetadataValidationRepository.class);
+
+					boolean isInvalid =
+							(metadataValidationRepository.count(MetadataValidationSpecs.isInvalidAndRequiredForMetadata(Integer.parseInt(id))) > 0);
+
+					if (isInvalid) {
+						operationAllowedRepo.deleteAll(where(hasMetadataId(id)).and(hasGroupId(ReservedGroup.all.getId())));
+					}
+					dataMan.indexMetadata(id, true);
```

Comment:
Maybe you could only index once if there is forceValidation (line 161) + automaticUnpublish.